src/schema/posts.ts

```diff
@@ -2344,19 +2345,22 @@ export const resolvers: IResolvers<unknown, BaseContext> = traceResolvers<
       ctx: AuthContext,
       info,
     ): Promise<GQLPost> => {
-      await Promise.all([
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      const [post, __, ___] = await Promise.all([
         ctx.con.getRepository(Post).findOneByOrFail({ id }),
```

Comment:
Should we select only the necessary fields here?

Reply:
we can, didn't want to touch it as no changes needed.

Reply:
Oh we actually can't because it queries Post, but can also satisfy SharedPost we can't query those fields.

Reply:
If you use findOneOrFail instead By it should allow that. We should avoid selecting all fields. I know it wasn't you but since you're changing it already, let's adjust please

Reply:
It's not the by that's the issue.
It's the fact we take `Post` repo, but we need a field from the child entity (`SharedPost`)

However can't make this SharedPost as it needs to satisfy all types.
Hope this makes sense?

Reply:
Done :)